src/ORM/FieldType/DBDate.php

```diff
@@ -67,6 +67,9 @@ protected function parseDate($value)
         } else {
             // Convert US date -> iso, fix y2k, etc
             $value = $this->fixInputDate($value);
+            if (is_null($value)) {
```

Comment:
Can you please add a test that `0-0-0` creates a null value?

Reply:
I've added a couple of tests for this case :)